core/src/main/scala/dagr/core/tasksystem/Dependable.scala

```diff
@@ -41,17 +50,45 @@ trait Dependable {
   def !=> (other: Dependable) : Unit
 
   /** Returns an object that can be used to manage dependencies that apply to this and the other Dependable. */
-  def :: (other: Dependable) : Dependable = new DependencyGroup(this, other)
+  def :: (other: Dependable) : Dependable = (this, other) match {
+    case (EmptyDependable, d) => d
+    case (d, EmptyDependable) => d
+    case _                    => new DependencyGroup(other, this)
+  }
+
+  /** Returns an object that can be used to manage dependencies that apply to this and the other Dependable. */
+  def :: (other: Option[Dependable]) : Dependable = other match {
+    case None    => this
+    case Some(d) => this :: d
+  }
 
   /** Abstract method that must be implemented to return all Tasks associated with this Dependable. */
   private[tasksystem] def toTasks: Traversable[Task]
 }
 
+/** An object that can be implicitly converted to from a None when using Option[Dependable]. */
+object EmptyDependable extends Dependable {
+  /** Converts an Option[Dependable] to a Dependable when needed. */
+  implicit def optionDependableToDependable(maybe: Option[Dependable]): Dependable = maybe match {
```

Comment:
I am wondering if this is the right place for this method, since it doesn't have any outward facing references to EmptyDependable.